api/src/main/resources/templates/pdfviewer/pdfviewer.vm

```diff
@@ -35,7 +35,11 @@
   ## parameter, since an attachment picker is used, or by using both the file and document parameters, for macros added
   ## in xwiki versions before 11.5.
   #if ($file.startsWith('http://') || $file.startsWith('https://'))
-    #set ($url = $file)
+     #set ($attachRef = $services.model.resolveAttachment($file, "resource/standardURL"))
+     #set ($docRef = $services.model.resolveDocument('', 'default', $attachRef))
```

Comment:
You should just get this document from the `$attachRef`, no need to resolve it
Check https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-ui/src/main/resources/XWiki/Lightbox/AttachmentMetaDataService.xml#L42 for an example.
When you don't know what methods you can use, check the class with `.class.name`, which in your case is [AttachmentReference](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/main/java/org/xwiki/model/reference/AttachmentReference.java#L127), and check the methods it offers.